lib/ansible/modules/cloud/amazon/ec2_vpc_igw.py

```diff
@@ -80,128 +83,205 @@
 '''
 
 try:
-    import boto.ec2
-    import boto.vpc
-    from boto.exception import EC2ResponseError
-    HAS_BOTO = True
+    import botocore
 except ImportError:
-    HAS_BOTO = False
-
-from ansible.module_utils.basic import AnsibleModule
-from ansible.module_utils.ec2 import AnsibleAWSError, connect_to_aws, ec2_argument_spec, get_aws_connection_info
+    pass # Handled by AnsibleAWSModule
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import (
+    AnsibleAWSError,
+    AWSRetry,
+    boto3_conn,
+    ec2_argument_spec,
+    get_aws_connection_info,
+    camel_dict_to_snake_dict,
+    boto3_tag_list_to_ansible_dict,
+    ansible_dict_to_boto3_filter_list,
+    ansible_dict_to_boto3_tag_list,
+    compare_aws_tags
+)
 from ansible.module_utils.six import string_types
-
+import time
 
 class AnsibleIGWException(Exception):
     pass
 
 
-def get_igw_info(igw):
-    return {'gateway_id': igw.id,
-            'tags': igw.tags,
-            'vpc_id': igw.vpc_id
-            }
+class AnsibleEc2Igw(object):
 
+    def __init__(self, module, results):
+        self._module = module
+        self._results = results
+        self._connection = None
+        self._check_mode = self._module.check_mode
 
-def get_resource_tags(vpc_conn, resource_id):
-    return dict((t.name, t.value) for t in
-                vpc_conn.get_all_tags(filters={'resource-id': resource_id}))
+    def connect(self):
```

Comment:
You don't need to access get_aws_connection_info and boto3_conn directly and any errors with the connection should be handled by boto3_conn. After creating the module by `module = AnsibleAWSModule(...` the connection can be created with `module.client('ec2')`.